### PR TITLE
Feature/updates for rc2

### LINKED ISSUE #123: Duplicate packages being build with different hashes because of different Autoconf provider
See https://github.com/spack/spack/issues/29930

I found this on the NOAA Cloud platforms. Pkgconf is defined as an external, but it's not being used in all instances, and `pkg-config` is being used, resulting in different hashes for NetCDF. My solution was to set `pkg-config` to `buildable: False`.

### LINKED ISSUE #162: create mirror CI test failing
https://github.com/NOAA-EMC/spack-stack/runs/6663667075?check_suite_focus=true

It's still looking for `create.py`

### LINKED ISSUE #183: create-spack-mirror CI tests failing 06/06/2022
The create-spack-mirror CI tests are now (06/06/2022) failing with
```
==> Warning: included configuration files should be updated manually [files=${SPACK_STACK_DIR}/configs/common/modules.yaml, ${SPACK_STACK_DIR}/configs/common/config.yaml, site, packages.yaml]
==> Error: Invalid url format from url: ${SPACK_STACK_DIR}/cache/source_cache
Error: Process completed with exit code 3.
```

### LINKED ISSUE #206: py-pillow build error on Orion with gnu-10.2.0 (`gcc: error: unrecognized command-line option '-R'`) - update py-setuptools to 59.4.0 or remove version 57.4.0 from common packages.yaml
I think this error comes from using an external and very old `m4`. Trying to remove the external `m4` from the orion config and build again.

**If** that works, then also try without the external `m4` with Intel. (The intel build didn't have any problem with it.)

### LINKED ISSUE #211: ecflow not declared as a Python package
I'm testing the newly created stack on an AMI. I loaded all the modules needed for building fv3-bundle and running ewok. Running `import ecflow` gives me the error: 
```
ModuleNotFoundError: No module named 'ecflow'
```
It looks like ecflow is not declared as a Python package in spack and so is not set in `PYTHONPATH` correctly.

### PR DESCRIPTION
**Pull request by climbfuji**
A number of updates for spack-stack-1.0.0 release candidate 2:
- Update CI tests to use `jedi-ufs-all` template
- Fix `create-spack-mirror` CI test
- Add `yaksa` modulefile to blacklist
- Do not pin version for `py-setuptools`
- Update Discover site config
- Update Orion GNU site config
- By default, use environment modules (`tcl/tk`) instead of `lmod/lua` on AWS Ubuntu and Redhat
- Extend `jedi-ufs-all` template
- Update `skylab-1.0.0` template
- Add modulefile templates for `qt5` and `ecflow`
- Update documentation, especially instructions for building `ecflow`

Fixes #162 (see https://github.com/climbfuji/spack-stack/runs/6943697862?check_suite_focus=true)

Fixes https://github.com/NOAA-EMC/spack-stack/issues/211

Fixes https://github.com/NOAA-EMC/spack-stack/issues/213

Fixes https://github.com/NOAA-EMC/spack-stack/issues/123

Waiting on https://github.com/NOAA-EMC/spack/pull/106

CI tests all passed except for macos-apple-clang build (for some weird reason, the same test passed before - rerunning now without parallel spack installs). Rerunning macos-apple-clang now manually, check https://github.com/climbfuji/spack-stack/runs/6977275667?check_suite_focus=true for the progress.

### CODE REVIEW

**Comment by climbfuji** [comment]
@kgerheiser CI tests all pass, finally (https://github.com/NOAA-EMC/spack-stack/pull/214/commits/2934b243870563d0ff31509581487ed640a2eda2) - ready for merging from my side (first spack, then spack-stack).

Update. Looks like my last cleanup of the macOS CI build re-introduced the curl build error. I want to ignore this for the PR, as long as all the others pass, and fix it in a next PR, which has a few minor documentation updates that are not included here.

File: configs/sites/discover/packages.yaml
Comment (climbfuji) [inline configs/sites/discover/packages.yaml]: Note. This junk will be removed later if not in a last minute doc push to this PR.

File: configs/sites/discover/packages.yaml
Comment (climbfuji) [inline configs/sites/discover/packages.yaml]: This will be updated in a follow-up PR or in a last minute push to this PR.

File: configs/sites/discover/packages.yaml
Comment (climbfuji) [inline configs/sites/discover/packages.yaml]: This will be removed in a follow-up PR or in a last minute push to this PR.

File: doc/source/Prerequisites.rst
Comment (climbfuji) [inline doc/source/Prerequisites.rst:109]: This documentation will be updated and completed over the next week

**Comment by climbfuji** [comment]
All other CI tests passed. This is ready to merge once the spack PR was merged, the submodule pointer updated, and the `.gitmodules` change reverted.

File: .github/actions/setup-spack-stack/action.yaml
Comment (kgerheiser) [inline .github/actions/setup-spack-stack/action.yaml:294]: What makes it unreliable? I used this technique myself after seeing it here and it worked fine.

File: configs/common/packages.yaml
Comment (kgerheiser) [inline configs/common/packages.yaml]: Thanks!

File: .github/actions/setup-spack-stack/action.yaml
Comment (climbfuji) [reply .github/actions/setup-spack-stack/action.yaml:294]: What happened a few times was that the install step was skipped, as if it ran the query if the jobs are still running faster than they appeared in the `ps` output. I thought this shouldn't be possible, but simply repeating the same run worked. Could probably be solved by adding a little sleep call in between?

File: configs/common/packages.yaml
Comment (kgerheiser) [reply configs/common/packages.yaml]: Actually, I would make this `pkgconf:buildable:false`. As far as I can tell, `pkg-config` is available on every platform as an external and we want to disable mixing the two.

Comment (climbfuji) [reply configs/common/packages.yaml]: Should we mark both as buildable: False?

File: configs/common/packages.yaml
Comment (climbfuji) [inline configs/common/packages.yaml:214]: I think we can remove that comment as well.

File: configs/common/packages.yaml
Comment (climbfuji) [reply configs/common/packages.yaml]: Done.

**Review by kgerheiser — APPROVED** [review]

File: configs/common/packages.yaml
Comment (kgerheiser) [inline configs/common/packages.yaml]: One of them should be buildable. An external `pkg-config` is generally available on these HPC systems so that's why I say go for that and leave `pkgconf` as not buildable.

**Comment by climbfuji** [comment]
If we think that pkg-config is generally available, then it won’t hurt to turn building off for both. I’ll try that for rc2, and if it doesn’t work across all platforms I’ll remove the `pkg-config` entry.

**Comment by kgerheiser** [comment]
macOS does not have a `pkgconfig` provider built in

**Comment by climbfuji** [comment]
> macOS does not have a `pkgconfig` provider built in

Ok, so let me change it right away then. Thanks!

**Pull request merged**
